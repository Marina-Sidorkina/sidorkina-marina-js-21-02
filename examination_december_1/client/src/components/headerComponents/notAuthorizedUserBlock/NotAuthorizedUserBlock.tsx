import React from 'react';
import { Link } from 'react-router-dom';
import '../../../locale/i18next';
import { useTranslation } from 'react-i18next';
import styles from './NotAuthorizedUserBlock.module.scss';

const NotAuthorizedUserBlock = () => {
  const { t } = useTranslation();

  return (
    <div className={styles.userBlock}>
      <div className={styles.item}>
        <Link className={styles.link} to="/login">
          <span className={styles.signIn}>
            { t('authorization.signIn', {}) }
          </span>
        </Link>
      </div>
      <div className={styles.item}>
        <Link className={styles.link} to="/registration">
          <span className={styles.register}>
            { t('authorization.signUp', {}) }
          </span>
        </Link>
      </div>
    </div>
  );
};

export default NotAuthorizedUserBlock;
