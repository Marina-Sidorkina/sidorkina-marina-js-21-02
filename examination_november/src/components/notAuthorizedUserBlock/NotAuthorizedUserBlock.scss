.default-user-block {
  display: flex;
  align-items: center;
  height: 50px;

  &__item {
    display: flex;
    align-items: center;
    height: 40px;

    &:first-child {
      border-right: 1px solid lightgray;
      padding-right: 15px;
    }

    &:last-child {
      padding-left: 15px;
    }
  }

  &__signIn,
  &__register {
    cursor: pointer;

    &:hover {
      opacity: 0.7;
    }
  }

  @media (max-width: 769px) {
    align-items: flex-start;
    flex-direction: column;
    width: 150px;
    height: 100%;
    padding-top: 10px;

    &__item {
      box-sizing: border-box;
      justify-content: flex-end;
      height: 55px;
      width: 150px;
      font-size: 20px;

      &:first-child {
        border-right: none;
        padding-right: 15px;
      }

      &:last-child {
        padding-right: 15px;
      }
    }
  }
}
