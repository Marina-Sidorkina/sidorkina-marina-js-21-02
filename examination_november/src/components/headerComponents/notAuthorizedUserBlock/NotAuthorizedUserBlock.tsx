import React from 'react';
import './NotAuthorizedUserBlock.scss';
import { Link } from 'react-router-dom';

const NotAuthorizedUserBlock = () => (
  <div className="default-user-block">
    <div className="default-user-block__item">
      <Link className="default-user-block__link" to="/login">
        <span className="default-user-block__signIn">Вход</span>
      </Link>
    </div>
    <div className="default-user-block__item">
      <Link className="default-user-block__link" to="/registration">
        <span className="default-user-block__register">Регистрация</span>
      </Link>
    </div>
  </div>
);

export default NotAuthorizedUserBlock;
