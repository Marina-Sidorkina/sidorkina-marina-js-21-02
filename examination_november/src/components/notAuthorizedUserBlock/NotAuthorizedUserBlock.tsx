import React from 'react';
import './NotAuthorizedUserBlock.scss';

const NotAuthorizedUserBlock = () => (
  <div className="default-user-block">
    <div className="default-user-block__item">
      <span className="default-user-block__signIn">Вход</span>
    </div>
    <div className="default-user-block__item">
      <span className="default-user-block__register">Регистрация</span>
    </div>
  </div>
);

export default NotAuthorizedUserBlock;
