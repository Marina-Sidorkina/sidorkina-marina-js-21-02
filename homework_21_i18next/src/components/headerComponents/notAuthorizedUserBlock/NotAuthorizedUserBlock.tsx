import React from 'react';
import './NotAuthorizedUserBlock.scss';
import { Link } from 'react-router-dom';
import '../../../locale/i18next';
import { useTranslation } from 'react-i18next';

const NotAuthorizedUserBlock = () => {
  const { t } = useTranslation();

  return (
    <div className="default-user-block">
      <div className="default-user-block__item">
        <Link className="default-user-block__link" to="/login">
          <span className="default-user-block__signIn">
            { t('authorization.signIn', {}) }
          </span>
        </Link>
      </div>
      <div className="default-user-block__item">
        <Link className="default-user-block__link" to="/registration">
          <span className="default-user-block__register">
            { t('authorization.signUp', {}) }
          </span>
        </Link>
      </div>
    </div>
  );
};

export default NotAuthorizedUserBlock;
