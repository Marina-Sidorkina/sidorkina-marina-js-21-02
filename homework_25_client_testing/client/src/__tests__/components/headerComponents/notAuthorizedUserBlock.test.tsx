import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { configure, shallow } from 'enzyme';
import NotAuthorizedUserBlock from '../../../components/headerComponents/notAuthorizedUserBlock/NotAuthorizedUserBlock';

configure({ adapter: new Adapter() });

describe('NotAuthorizedUserBlock component test', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = shallow(<NotAuthorizedUserBlock />);
  });

  test('should render NotAuthorizedUserBlock', () => {
    expect(wrapper.length).toEqual(1);
  });

  test('should render NotAuthorizedUserBlock', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
