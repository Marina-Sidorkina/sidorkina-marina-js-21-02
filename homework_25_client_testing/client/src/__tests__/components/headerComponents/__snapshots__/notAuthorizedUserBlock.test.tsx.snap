// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotAuthorizedUserBlock component test should render NotAuthorizedUserBlock 1`] = `
<div
  className="userBlock"
>
  <div
    className="item"
  >
    <Link
      className="link"
      to="/login"
    >
      <span
        className="signIn"
      >
        Sign in
      </span>
    </Link>
  </div>
  <div
    className="item"
  >
    <Link
      className="link"
      to="/registration"
    >
      <span
        className="register"
      >
        Sign up
      </span>
    </Link>
  </div>
</div>
`;
